src/zope/testrunner/tb_format.py

```diff
@@ -20,6 +20,31 @@
 import zope.testrunner.feature
 
 
+# Python 3.5 compatibility
+if not hasattr(traceback, "_iter_chain"):
+    def _iter_chain(exc, custom_tb=None, seen=None):
+        if seen is None:
+            seen = set()
+        seen.add(exc)
+        its = []
+        cause = exc.__cause__
+        if cause is not None and cause not in seen:
+            its.append(_iter_chain(cause, None, seen))
+            its.append([(traceback._cause_message, None)])
+        else:
+            context = exc.__context__
+            if context is not None and context not in seen:
+                its.append(_iter_chain(context, None, seen))
+                its.append([(traceback._context_message, None)])
+        its.append([(exc, custom_tb or exc.__traceback__)])
+        # itertools.chain is in an extension module and may be unavailable
+        for it in its:
+            for x in it:
+                yield x
+
+    traceback._iter_chain = _iter_chain
```

Comment:
Would it be better to avoid monkey-patching a stdlib module, and instead make `_iter_chain` a local function? Something like:
```python
try:
    _iter_chain = traceback._iter_chain
except AttributeError:
    # Python 3.5
    def _iter_chain(...):

...
def format_exception(...):
  if chain:
       values = _iter_chain(v, tb)
```